test/statetest/statetest_runner.cpp

```diff
@@ -0,0 +1,47 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../state/mpt_hash.hpp"
+#include "../state/rlp.hpp"
+#include "statetest.hpp"
+#include <gtest/gtest.h>
+
+namespace evmone::state
+{
+/// Defines how to RLP-encode a Log. This is only needed to compute "logs hash".
+inline bytes rlp_encode(const Log& log)
+{
+    return rlp::encode_tuple(log.addr, log.topics, log.data);
+}
+}  // namespace evmone::state
+
+namespace evmone::test
+{
+void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
+{
+    for (const auto& [rev, cases] : test.cases)
+    {
+        for (size_t case_index = 0; case_index != cases.size(); ++case_index)
+        {
+            SCOPED_TRACE(std::string{evmc::to_string(rev)} + '/' + std::to_string(case_index));
+            // if (rev != EVMC_FRONTIER)
```

Comment:
What are these comments?

Reply:
There are for debugging. You can narrow a test to single revision and/or case.